src/replicache.ts

```diff
@@ -55,10 +56,26 @@ export type Poke = {
   pullResponse: PullResponse;
 };
 
+const enum MutationRecoveryStatus {
+  NO_MUTATIONS_TO_RECOVER,
+  SUCCESS,
+  ERROR,
+}
+
+type MutationRecoveryResult = {
+  status: MutationRecoveryStatus;
+  /**
+   * When `status` is `SUCCESS`, the new `ClientMap` state, otherwise
```

Comment:
can you use a discriminated union here?

Reply:
decided to simplify and not fail fast on connectivity issues, this led to not needing this more complicated return structure.